doc/user/http-frontend-query.rst

```diff
@@ -0,0 +1,423 @@
+
+..  list-table::
+    :widths: 10 90
+    :header-rows: 0
+
+    * - ``GET``
+      - ``/query-async/status/:queryId``
+
+Where:
+
+``queryId`` : *number*
+  The required unique identifier of the previously submitted query.
+
+If the query identifier is not valid, the service will report an error in the response object.
+For example, consider the following request:
+
+.. code-block:: bash
+
+    curl -k 'https://localhost:4041/query-async/status/123?version=38' -X GET
+
+It might result in the following response:
+
+.. code-block:: json
+
+  { "success" : 0,
+    "error" :   "failed to obtain info for queryId=123,
+                 ex: Czar::getQueryInfo Unknown user query, err=,
+                 sql=SELECT status,messageTable,resultQuery FROM QInfo WHERE queryId=123",
+    "error_ext" : {},
+  }
+
+If the query identifier is valid then the following object will be returned:
+
+.. code-block::
+
+    {   "success" : 1,
+        ...
+        "status" : {
+            "queryId" :         <number>,
+            "status" :          <string>,
+            "totalChunks" :     <number>,
+            "completedChunks" : <number>,
+            "queryBeginEpoch" : <number>,
+            "lastUpdateEpoch" : <number>
+        }
+    }
+
+Where the ``status`` is an object that has following attributes:
+
+``queryId`` : *number*
+  The unique identifier of the previously submitted query.
+
+``status`` : *string*
+  The current status of the query can have one of the following values:
+
+  - ``EXECUTING`` - The query processing is still in progress.
+  - ``COMPLETED`` - The query has been completed.
+  - ``FAILED`` - The query failed.
+  - ``ABORTED`` - The query was aborted either:
+
+    - explicitly by a user using the query cancellation REST service explained in the document.
+    - or, implicitly by Qserv if the intermediate result set of the query exceeds the large result
+      limit (which is configured by the Qserv administrators).
+    - or, implicitly when the query processing service was restarted due to some failure or by
+      Qserv administrators.
+
+``totalChunks`` : *number*
+  The total number of so-called "chunks" (spatial shards used for splitting the large tables in Qserv
+  into smaller sub-tables to be distributed across worker nodes of Qserv).
+
+``completedChunks`` : *number*
+  The number of chunks that have been processed by Qserv so far. The value of this parameter varies
+  from ``0`` to the maximum number reported in the attribute ``totalChunks``.
+
+``queryBeginEpoch`` : *number*
+  The 32-bit number representing the start time of the query expressed in seconds since the UNIX *Epoch*.
+
+``lastUpdateEpoch`` : *number*
+  The 32-bit number represents the last time when the query status was recorded/updated by the Qserv
+  internal monitoring system. The timestamp is the number of seconds since the UNIX *Epoch*.
+  The service returns a value of ``0`` if either of the following is true:
```

Comment:
The UNIX Epoch's "Year 2k Problem is looming". Unfortunately, there is nothing I can do here since the timestamps are produced by the UNIX library functions. Luckily, the telescope begins making observations in the year 2025 and finishes 10 years later (2035), well before the "deadline". Hopefully, I will retire before that 😸